src/main/java/org/deepsymmetry/beatlink/AnnouncementSocketConnection.java

```diff
@@ -0,0 +1,294 @@
+    private static final Logger logger = LoggerFactory.getLogger(AnnouncementSocketConnection.class);
+
+    public static final AnnouncementSocketConnection connectionsManager = new AnnouncementSocketConnection();
+
+    /**
+     * The socket used to receive device status packets while we are active.
+     */
+    private final AtomicReference<DatagramSocket> announcementSocket = new AtomicReference<DatagramSocket>();
+
+    /**
+     * Track when we started listening for announcement packets.
+     */
+    private static final AtomicLong startTime = new AtomicLong();
+
+    /**
+     * Maintain a set of addresses from which device announcements should be ignored. The {@link VirtualCdj} will add
+     * its socket to this set when it is active so that it does not show up in the set of devices found on the network.
+     */
+    private final Set<InetAddress> ignoredAddresses =
+            Collections.newSetFromMap(new ConcurrentHashMap<InetAddress, Boolean>());
+
+    /**
+     * The port to which devices broadcast announcement messages to report their presence on the network.
+     */
+    public static final int ANNOUNCEMENT_PORT = 50000;
+
+    /**
+     * Check whether we are presently posing as a virtual CDJ and receiving device status updates.
+     *
+     * @return true if our socket is open, sending presence announcements, and receiving status packets
+     */
+    public boolean isRunning() {
+        return announcementSocket.get() != null;
+    }
+    public static AnnouncementSocketConnection getInstance() {
+        return connectionsManager;
+    }
+
+    /**
+     * Send a master changed announcement to all registered master listeners.
+     *
+     * @param announcement the message announcing the new announcements
+     */
+    private void deliverDeviceAnnouncement(final DeviceAnnouncement announcement) {
+        for (final AnnouncementListener listener : getDeviceAnnouncementListeners()) {
+            try {
+                listener.handleDeviceAnnouncement(announcement);
+            } catch (Throwable t) {
+                logger.warn("Problem delivering announcement to listener", t);
+            }
+        }
+    }
+
+    /**
+     * Start listening for device announcements and keeping track of the DJ Link devices visible on the network.
+     * If already listening, has no effect.
+     *
+     * This should only be called by classes that use the announcement socket (eg: DeviceFinder/VirtualRekordbox).
+     *
+     * @throws SocketException if the socket to listen on port 50000 cannot be created
+     */
+    public synchronized void start() throws SocketException {
+        if (!isRunning()) {
+            // This needs to happen first as this will be how users of this
+            announcementSocket.set(new DatagramSocket(ANNOUNCEMENT_PORT));
+            startTime.set(System.currentTimeMillis());
+            deliverLifecycleAnnouncement(logger, true);
```

Comment:
As a `LifecycleParticipant`, we need to deliver a lifecycle announcement that we are starting here. See the corresponding section of `DeviceListener`.

Reply:
Isn't that whats happening on this line?